### DISCUSSION TITLE
Listing table columns

### DISCUSSION THREAD

**Discussion by RicardoE105** [supabase/supabase#4338] [Questions]
How do I return all the columns in a table?

**Comment by GaryAustin1** [comment]
From SQL 
select column_name, data_type
    from INFORMATION_SCHEMA.COLUMNS 
    where table_name = 'tablename';

From JS you could fetch a single record with select * and process the returned json for object keys.

**Reply by RicardoE105** [reply]
I'm using the SDK as shown below but I get the error: "message: 'relation "public.information_schema.columns" does not exist'". @GaryAustin1 what Am I missing?

```
	const { data, error } = await supabase
		.from('information_schema.columns')
		.select('*')
		.eq('table_name', 'testing')
```

**Reply by GaryAustin1** [reply]
If you want to access other than the public schema you have to set up another client like:

const supabase = createClient('https://xyzcompany.supabase.co', 'public-anon-key', {  schema: 'information_schema' })

The JS client by default only accesses the public schema.

BUT try just doing .from('testing').select('*').limit(1) if you have a row in the table.  Then from the returned object you can extract the column names with Object.keys(data[0]).

**Reply by RicardoE105** [reply]
Thanks very much for your help @GaryAustin1 

The second approach will not work because I need to know which columns are the primary key, that is why I need access to the information_schema

Tried the first approach but got the following error:

{ message: 'The schema must be one of the following: public, storage' }

**Reply by GaryAustin1** [reply]
Ah, so I guess privileged schemas still can't go thru the js handler.  You will need to write a function using the sql code I provided, then return your columns in an rpc call.

**Reply by RicardoE105** [reply]
Thanks @GaryAustin1

**Reply by RicardoE105** [reply]
I was able to get the table metadata (tables and columns), making a GET HTTP request to `https://yoursubdomai/rest/v1/`.
